services/analytics/object_details_test.go

```diff
@@ -0,0 +1,295 @@
+		rm models.Reporter
+		mm models.Metrics
+	}
+	type args struct {
+		ctx context.Context
+		in  *qanpb.QueryExampleRequest
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *qanpb.QueryExampleReply
+		wantErr bool
+	}{
+		{
+			"no_period_start_from",
+			fields{rm: rm, mm: mm},
+			args{
+				context.TODO(),
+				&qanpb.QueryExampleRequest{
+					PeriodStartTo: &timestamp.Timestamp{Seconds: t2.Unix()},
+					GroupBy:       "queryid",
+					FilterBy:      "B305F6354FA21F2A",
+					Limit:         5,
+				},
+			},
+			nil,
+			true,
+		},
+		{
+			"no_period_start_to",
+			fields{rm: rm, mm: mm},
+			args{
+				context.TODO(),
+				&qanpb.QueryExampleRequest{
+					PeriodStartFrom: &timestamp.Timestamp{Seconds: t1.Unix()},
+					GroupBy:         "queryid",
+					FilterBy:        "B305F6354FA21F2A",
+					Limit:           5,
+				},
+			},
+			nil,
+			true,
+		},
+		{
+			"no_group",
+			fields{rm: rm, mm: mm},
+			args{
+				context.TODO(),
+				&qanpb.QueryExampleRequest{
+					PeriodStartFrom: &timestamp.Timestamp{Seconds: t1.Unix()},
+					PeriodStartTo:   &timestamp.Timestamp{Seconds: t2.Unix()},
+					FilterBy:        "B305F6354FA21F2A",
+					Limit:           5,
+				},
+			},
+			&want,
+			false,
+		},
+		{
+			"no_limit",
+			fields{rm: rm, mm: mm},
+			args{
+				context.TODO(),
+				&qanpb.QueryExampleRequest{
+					PeriodStartFrom: &timestamp.Timestamp{Seconds: t1.Unix()},
+					PeriodStartTo:   &timestamp.Timestamp{Seconds: t2.Unix()},
+					GroupBy:         "queryid",
+					FilterBy:        "B305F6354FA21F2A",
+				},
+			},
+			&want,
+			false,
+		},
+		{
+			"invalid_group_name",
+			fields{rm: rm, mm: mm},
+			args{
+				context.TODO(),
+				&qanpb.QueryExampleRequest{
+					PeriodStartFrom: &timestamp.Timestamp{Seconds: t1.Unix()},
+					PeriodStartTo:   &timestamp.Timestamp{Seconds: t2.Unix()},
+					GroupBy:         "invalid_group_name",
+					FilterBy:        "B305F6354FA21F2A",
+				},
+			},
+			nil,
+			true,
+		},
+		{
+			"not_found",
+			fields{rm: rm, mm: mm},
+			args{
+				context.TODO(),
+				&qanpb.QueryExampleRequest{
+					PeriodStartFrom: &timestamp.Timestamp{Seconds: t1.Unix()},
+					PeriodStartTo:   &timestamp.Timestamp{Seconds: t2.Unix()},
+					GroupBy:         "queryid",
+					FilterBy:        "unexist",
+				},
+			},
+			&want,
+			false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &Service{
+				rm: tt.fields.rm,
+				mm: tt.fields.mm,
+			}
+			got, err := s.GetQueryExample(tt.args.ctx, tt.args.in)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Service.GetQueryExample() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			tt.want = nil
+			expectedData(t, got, &tt.want, "../../test_data/GetQueryExample_"+tt.name+".json")
+
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("Service.GetQueryExample() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestService_GetMetrics(t *testing.T) {
```

Comment:
You could use subtests there:
```go
func TestService(t *testing.T) {
  t.Run("GetMetrics", func(t *testing.T) {
    // …
  })
```

Reply:
Yes, thank you, I know. But it generates by https://github.com/cweill/gotests and looks less repetitive.